tests/algorithms/test_algorithm_resumption.py

```diff
@@ -42,7 +42,7 @@ def test_algorithm_resumption(
         'max_duration': '2ep',
         'save_filename': 'ep{epoch}-rank{rank}',
         'train_subset_num_batches': 2,
-        'precision': 'fp32',
+        'precision': 'amp',
```

Comment:
Would it work to add a seperate test case to test that the grad scaler state is being properly restored? (I presume this change was made here to test that, but would prefer to have a separate test in `test_trainer` if that's not too much work.)

Reply:
Actually, this is reverting a change made in https://github.com/mosaicml/composer/pull/1369.